lib/prop/limiter.rb

```diff
@@ -46,7 +46,8 @@ def before_throttle(&blk)
       #
       # Raises Prop::RateLimited if the number if the threshold for this handle has been reached
       def configure(handle, defaults)
-        raise ArgumentError.new("Invalid threshold setting") unless defaults[:threshold].to_i > 0
+        raise ArgumentError.new("Invalid threshold setting") unless defaults[:threshold].is_a? Integer
+        raise ArgumentError.new("Invalid threshold setting") unless defaults[:threshold].to_i >= 0
```

Comment:
seems weird that we have the same validation logic twice ?

Reply:
Is this better if 
```
raise ArgumentError.new("Invalid threshold setting") unless defaults[:threshold].is_a? Integer
raise ArgumentError.new("Invalid threshold setting") unless defaults[:threshold] >= 0
```
Or we can merge the conditions together but it's a bit long
```
raise ArgumentError.new("Invalid threshold setting") unless defaults[:threshold].is_a? Integer && defaults[:threshold] >= 0
```

And we notice that a string converted to an integer when comparing to 0 is true, e.g: 
``` "wobble".to_i == 0  // true```

Reply:
Alternatively this would validate the params:
```
        # validate defaults by passing them into Prop::Options.build:
        Prop::Options.build(key: nil, params: {}, defaults: defaults)
```

Reply:
I tried it out, it's looked much cleaner than before.